### PR TITLE
Simpletests to ensure filter is working properly.

### PR DESCRIPTION
**Pull request by cs-shadow**
d.o issue: https://www.drupal.org/node/2290277

### CODE REVIEW

**Comment by jibran** [comment]
I think this can only be tested by https://www.drupal.org/node/2288871#comment-8895215

File: src/Tests/EntityEmbedFilterTest.php [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+  }
+
+  protected function setUp() {
+    parent::setUp();
+
+    // Create a page content type.
+    $this->drupalCreateContentType(array('type' => 'page', 'name' => 'Basic page'));
+
+    // Create Filtered HTML text format and enable entity_embed filter.
+    $filtered_html_format = entity_create('filter_format', array(
```
Comment (jibran) [inline src/Tests/EntityEmbedFilterTest.php]: This should be full html. Drupal 8 don't support divs in filtered html.

File: src/Tests/EntityEmbedFilterTest.php [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    $settings = array();
+    $settings['type'] = 'page';
+    $settings['title'] = 'Test entity embed with entity-uuid and view-mode';
+    $settings['body'] = array(array('value' => $content));
+    $node = $this->drupalCreateNode($settings);
+
+    $html = $this->drupalGet('node/' . $node->id());
+
+    $this->assertText($this->embedContent, 'Embedded node exists in page.');
+    $this->assertNoText('This placeholder should not be rendered.', 'Placeholder does not appears in the output when embed is successful.');
```
Comment (jibran) [inline src/Tests/EntityEmbedFilterTest.php]: how about using `$content` instead of 'This placeholder should not be rendered.'

File: src/Tests/EntityEmbedFilterTest.php (hunk 1 above)
Comment (cs-shadow) [inline src/Tests/EntityEmbedFilterTest.php]: Umm, this $filtered_html_format is not exactly Drupal's restricted html format. Its a format that we are defining. And since we have not enabled the `filter_html` filter which restricts html tags, this format is essentially full html with the entity_embed filter enables.
I can verify that this works since I've tested in on my local installation.

File: src/Tests/EntityEmbedFilterTest.php (hunk 2 above)
Comment (cs-shadow) [inline src/Tests/EntityEmbedFilterTest.php]: Nice idea, I'll replace all such occurrences.

**Comment by cs-shadow** [comment]
@jibran, unless we've the travis-ci, we can test this on our local installations using the `run-tests.sh` script.

File: src/Tests/EntityEmbedFilterTest.php (hunk 1 above)
Comment (slashrsm) [inline src/Tests/EntityEmbedFilterTest.php]: Maybe rename this format to prevent confusion.

Comment (cs-shadow) [inline src/Tests/EntityEmbedFilterTest.php]: OK, what about `$full_html` since essentially its full html only?

Comment (slashrsm) [inline src/Tests/EntityEmbedFilterTest.php]: I'd go with something that is completely unrelated to filters that ship with core.

File: src/Tests/EntityEmbedFilterTest.php (hunk 2 above)
Comment (cs-shadow) [inline src/Tests/EntityEmbedFilterTest.php]: (Somehow my earlier comment got removed, so posting again).
This might not be a good idea, since assertText excepts plain text as argument and `$content` is a HTML string, so I think we should not use `$content` here.

File: src/Tests/EntityEmbedFilterTest.php (hunk 1 above)
Comment (cs-shadow) [inline src/Tests/EntityEmbedFilterTest.php]: OK, naming stuff is always difficult for me.

Comment (cs-shadow) [inline src/Tests/EntityEmbedFilterTest.php]: Renamed to `custom_html` format.

File: src/Tests/EntityEmbedFilterTest.php (hunk 2 above)
Comment (slashrsm) [inline src/Tests/EntityEmbedFilterTest.php]: what about {code}strip_tags($content){code}?

Comment (cs-shadow) [inline src/Tests/EntityEmbedFilterTest.php]: Done.
